desc/experimental/_terpsichore.py

```diff
@@ -0,0 +1,872 @@
+    phips[0] = 0
+    phips[1:] = -Psi * np.ones((surfs - 1,)) / (2 * np.pi)
+
+    vp = file.createVariable("vp", np.float64, ("radius",))
+    vp.long_name = "dV/ds normalized by 4*pi^2, on half mesh"
+    vp.units = "m^3"
+    vp[1:] = grid_half.compress(data_half["V_r(r)"]) / (
+        8 * np.pi**2 * grid_half.compress(data_half["rho"])
+    )
+    vp[0] = 0
+
+    # R
+    rmnc = file.createVariable("rmnc", np.float64, ("radius", "mn_mode"))
+    rmnc.long_name = "cos(m*t-n*p) component of cylindrical R, on full mesh"
+    rmnc.units = "m"
+    if not eq.sym:
+        rmns = file.createVariable("rmns", np.float64, ("radius", "mn_mode"))
+        rmns.long_name = "sin(m*t-n*p) component of cylindrical R, on full mesh"
+        rmns.units = "m"
+    r1 = np.ones_like(eq.R_lmn)
+    r1[eq.R_basis.modes[:, 1] < 0] *= -1
+    m, n, x_mn = zernike_to_fourier(r1 * eq.R_lmn, basis=eq.R_basis, rho=r_full)
+    xm, xn, s, c = ptolemy_identity_rev(m, n, x_mn)
+    rmnc[:] = c
+    if not eq.sym:
+        rmns[:] = s
+
+    # Z
+    zmns = file.createVariable("zmns", np.float64, ("radius", "mn_mode"))
+    zmns.long_name = "sin(m*t-n*p) component of cylindrical Z, on full mesh"
+    zmns.units = "m"
+    if not eq.sym:
+        zmnc = file.createVariable("zmnc", np.float64, ("radius", "mn_mode"))
+        zmnc.long_name = "cos(m*t-n*p) component of cylindrical Z, on full mesh"
+        zmnc.units = "m"
+    z1 = np.ones_like(eq.Z_lmn)
+    z1[eq.Z_basis.modes[:, 1] < 0] *= -1
+    m, n, x_mn = zernike_to_fourier(z1 * eq.Z_lmn, basis=eq.Z_basis, rho=r_full)
+    xm, xn, s, c = ptolemy_identity_rev(m, n, x_mn)
+    zmns[:] = s
+    if not eq.sym:
+        zmnc[:] = c
+
+    # derived quantities (approximate conversion)
+
+    def fit(x):
+        y = fit_transform.fit(x)
+        return np.where(fit_transform.basis.modes[:, 1] < 0, -y, y)
+
+    # Jacobian
+    gmnc = file.createVariable("gmnc", np.float64, ("radius", "mn_mode_nyq"))
+    gmnc.long_name = "cos(m*t-n*p) component of Jacobian, on half mesh"
+    gmnc.units = "m"
+    if not eq.sym:
+        gmns = file.createVariable("gmns", np.float64, ("radius", "mn_mode_nyq"))
+        gmns.long_name = "sin(m*t-n*p) component of Jacobian, on half mesh"
+        gmns.units = "m"
+    m = fit_transform.basis.modes[:, 1]
+    n = fit_transform.basis.modes[:, 2]
+    # d(rho)/d(s) = 1/(2*rho)
+    data = (
+        (data_half["sqrt(g)"] / (2 * data_half["rho"]))
+        .reshape(
+            (grid_half.num_theta, grid_half.num_rho, grid_half.num_zeta), order="F"
+        )
+        .transpose((1, 0, 2))
+        .reshape((grid_half.num_rho, -1), order="F")
+    )
+    x_mn = np.zeros((surfs - 1, m.size))
+    for i in range(surfs - 1):
+        x_mn[i, :] = fit(data[i, :])
+    xm, xn, s, c = ptolemy_identity_rev(m, n, x_mn)
+    gmnc[0, :] = 0
+    gmnc[1:, :] = -c  # negative sign for negative Jacobian
+    if not eq.sym:
+        gmns[0, :] = 0
+        gmns[1:, :] = -s
```

Comment:
I thought we only need to negate one of these (cos)? since we negate the sign modes bc the pol. angle changed direction and we had fit it with the DESC poloidal angle, not the VMEC one, and we also negate entire g because the jacobian now has a sign change due to the reverse of the poloidal angle

so it is like gmnc = - c and gmns = - (-s)?

Reply:
maybe I am misunderstanding since I see we have this in the VMECIO.save as well.

Reply:
nvm, the missing - is in fit()